gradle-guide-error-prone/src/test/java/com/palantir/gradle/guide/errorprone/taskexecution/ArchiveOperationsFixesTest.java

```diff
@@ -0,0 +1,250 @@
+                abstract class AlreadyInjectedTask extends DefaultTask {
+                    private static final String IM_AT_THE_TOP_OF_THE_CLASS = "happy_squirrel.txt";
+                    @Inject
+                    protected abstract ArchiveOperations getArchiveOps();
+                    @TaskAction
+                    final void action() {
+                        FileTree fileTree = getArchiveOps().tarTree("happy-squirrel.tar");
+                    }
+                }
+                """);
+    }
+
+    @Test
+    void concrete_task_should_fix() {
+        testFix(
+                "ConcreteTask.java",
+                """
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.Plugin;
+                import org.gradle.api.Project;
+                import org.gradle.api.file.FileTree;
+                import org.gradle.api.tasks.TaskAction;
+                class ConcreteTask extends DefaultTask {
+                    @TaskAction
+                    final void action() {
+                        FileTree fileTree = getProject().tarTree("happy-squirrel.tar");
+                    }
+                }
+                """,
+                """
+                import javax.inject.Inject;
+                import org.gradle.api.DefaultTask;
+                import org.gradle.api.Plugin;
+                import org.gradle.api.Project;
+                import org.gradle.api.file.ArchiveOperations;
+                import org.gradle.api.file.FileTree;
+                import org.gradle.api.tasks.TaskAction;
+                abstract class ConcreteTask extends DefaultTask {
+                    @Inject
+                    protected abstract ArchiveOperations getArchiveOperations();
+                    @TaskAction
+                    final void action() {
+                        FileTree fileTree = getArchiveOperations().tarTree("happy-squirrel.tar");
+                    }
+                }
+                """);
+    }
+
+    @Test
+    void transitive_calls_to_getProject_copy_should_fail_in_taskActions() {
+        test(
+                """
+                import org.gradle.api.Action;
+                import org.gradle.api.Task;
+                abstract class CustomTaskAction implements Action<Task> {
+                    @Override
+                    public void execute(Task task) {
+                        // BUG: Diagnostic contains: Instead of `getProject().tarTree(...)`, do `ArchiveOperations#tarTree`
+                        task.getProject().tarTree("happy-squirrel.tar");
+                        bad_helper(task);
+                    }
+                    public void bad_helper(Task task) {
+                        System.out.println("I am a happy squirrel");
+                        // BUG: Diagnostic contains: Instead of `getProject().tarTree(...)`, do `ArchiveOperations#tarTree`
+                        task.getProject().tarTree("happy-squirrel.tar");
```

Comment:
I'm a bit confused, why doesn't this replace? I thought you had all the call graph stuff working?